From 7e83595e1aff4f25f0f4dcbdda29915ddbfdd8fa Mon Sep 17 00:00:00 2001
From: Mika Laitio <mika.laitio@amd.com>
Date: Wed, 15 Apr 2026 19:45:55 -0700
Subject: [PATCH 14/22] rocprofiler-sdk add ROCPROFILER_BUILD_ELFIO option

Add new build flag ROCPROFILER_BUILD_ELFIO which
is by default on to support the current way of building.

When the option is set off, the elfio is tried to find from
the system path instead of downloading it. This allows
therock to install the same elfio version that rocprofiler-sdk
uses and then patch it to support the gcc 15 to fix a following
build error:

therock/build/third-party/elfio/build_elfio/
    dist/include/elfio/elf_types.hpp:30:20:
    error: unknown type name 'uint16_t'
    9.5        30 | using Elf_Half   = uint16_t;

Later the same elfio version may be used also on some other
projects.

fixes for: https://github.com/ROCm/TheRock/issues/3517

Signed-off-by: Mika Laitio <mika.laitio@amd.com>
---
 ...k-add-ROCPROFILER_BUILD_ELFIO-option.patch | 101 ++++++++++++++++++
 1 file changed, 101 insertions(+)
 create mode 100644 patches/amd-mainline/rocm-systems/0007-rocprofiler-sdk-add-ROCPROFILER_BUILD_ELFIO-option.patch

diff --git a/patches/amd-mainline/rocm-systems/0007-rocprofiler-sdk-add-ROCPROFILER_BUILD_ELFIO-option.patch b/patches/amd-mainline/rocm-systems/0007-rocprofiler-sdk-add-ROCPROFILER_BUILD_ELFIO-option.patch
new file mode 100644
index 0000000..3dfb022
--- /dev/null
+++ b/patches/amd-mainline/rocm-systems/0007-rocprofiler-sdk-add-ROCPROFILER_BUILD_ELFIO-option.patch
@@ -0,0 +1,101 @@
+From 09ed3b285a3125be6847a96b3cf2337eb5ef34ab Mon Sep 17 00:00:00 2001
+From: Mika Laitio <mika.laitio@amd.com>
+Date: Tue, 31 Mar 2026 00:50:47 -0700
+Subject: [PATCH] rocprofiler-sdk add ROCPROFILER_BUILD_ELFIO option
+
+Add new build flag ROCPROFILER_BUILD_ELFIO which
+is by default on to support the current way of building.
+
+When the option is set off, the elfio is tried to find from
+the system path instead of downloading it. This allows
+therock to install the same elfio version that rocprofiler-sdk
+uses and then patch it to support the gcc 15 to fix a following
+build error:
+
+therock/build/third-party/elfio/build_elfio/
+    dist/include/elfio/elf_types.hpp:30:20:
+    error: unknown type name 'uint16_t'
+    9.5        30 | using Elf_Half   = uint16_t;
+
+Later the same elfio version may be used also on some other
+projects.
+
+fixes for: https://github.com/ROCm/TheRock/issues/3517
+
+Signed-off-by: Mika Laitio <mika.laitio@amd.com>
+---
+ .../cmake/rocprofiler_config_interfaces.cmake |  7 +++++-
+ .../cmake/rocprofiler_options.cmake           |  2 ++
+ .../rocprofiler-sdk/external/CMakeLists.txt   | 22 ++++++++++---------
+ 3 files changed, 20 insertions(+), 11 deletions(-)
+
+diff --git a/projects/rocprofiler-sdk/cmake/rocprofiler_config_interfaces.cmake b/projects/rocprofiler-sdk/cmake/rocprofiler_config_interfaces.cmake
+index eb1c340734..b716d5b823 100644
+--- a/projects/rocprofiler-sdk/cmake/rocprofiler_config_interfaces.cmake
++++ b/projects/rocprofiler-sdk/cmake/rocprofiler_config_interfaces.cmake
+@@ -310,9 +310,14 @@ endif()
+ #
+ # ----------------------------------------------------------------------------------------#
+ 
++if (NOT ROCPROFILER_BUILD_ELFIO)
++    # search system installed elfio
++    find_package(elfio CONFIG REQUIRED)
++endif()
++target_link_libraries(rocprofiler-sdk-elfio INTERFACE elfio::elfio)
++
+ # get_target_property(ELFIO_INCLUDE_DIR elfio::elfio INTERFACE_INCLUDE_DIRECTORIES)
+ # target_include_directories(rocprofiler-sdk-elfio SYSTEM INTERFACE ${ELFIO_INCLUDE_DIR})
+-target_link_libraries(rocprofiler-sdk-elfio INTERFACE elfio::elfio)
+ 
+ # ----------------------------------------------------------------------------------------#
+ #
+diff --git a/projects/rocprofiler-sdk/cmake/rocprofiler_options.cmake b/projects/rocprofiler-sdk/cmake/rocprofiler_options.cmake
+index c29311cc8c..efe3aedb3e 100644
+--- a/projects/rocprofiler-sdk/cmake/rocprofiler_options.cmake
++++ b/projects/rocprofiler-sdk/cmake/rocprofiler_options.cmake
+@@ -65,6 +65,8 @@ rocprofiler_add_option(ROCPROFILER_BUILD_PYBIND11
+                        "Enable building pybind11 library internally" ON)
+ rocprofiler_add_option(ROCPROFILER_BUILD_GOTCHA
+                        "Enable building gotcha library internally" ON)
++rocprofiler_add_option(ROCPROFILER_BUILD_ELFIO
++                       "Enable building elfio library internally" ON)
+ rocprofiler_add_option(ROCPROFILER_BUILD_YAML_CPP
+                        "Enable building yaml-cpp library internally" ON)
+ if(ROCPROFILER_BUILD_TESTS)
+diff --git a/projects/rocprofiler-sdk/external/CMakeLists.txt b/projects/rocprofiler-sdk/external/CMakeLists.txt
+index cab59c07bd..b430ce5ebd 100644
+--- a/projects/rocprofiler-sdk/external/CMakeLists.txt
++++ b/projects/rocprofiler-sdk/external/CMakeLists.txt
+@@ -200,17 +200,19 @@ target_link_libraries(
+     rocprofiler-sdk-perfetto
+     INTERFACE $<BUILD_INTERFACE:rocprofiler-sdk-perfetto-static-library>)
+ 
+-# ELFIO
+-rocprofiler_checkout_git_submodule(
+-    RECURSIVE
+-    RELATIVE_PATH external/elfio
+-    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+-    REPO_URL https://github.com/serge1/ELFIO.git
+-    REPO_BRANCH "Release_3.12")
++if (ROCPROFILER_BUILD_ELFIO)
++    # build elfio instead of using the system provided version
++    rocprofiler_checkout_git_submodule(
++        RECURSIVE
++        RELATIVE_PATH external/elfio
++        WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
++        REPO_URL https://github.com/serge1/ELFIO.git
++        REPO_BRANCH "Release_3.12")
+ 
+-set(ELFIO_BUILD_EXAMPLES OFF)
+-set(ELFIO_BUILD_TESTS OFF)
+-add_subdirectory(elfio EXCLUDE_FROM_ALL)
++    set(ELFIO_BUILD_EXAMPLES OFF)
++    set(ELFIO_BUILD_TESTS OFF)
++    add_subdirectory(elfio EXCLUDE_FROM_ALL)
++endif()
+ if(TARGET rocprofiler-sdk-elfio)
+     get_target_property(ELFIO_INCLUDE_DIR elfio::elfio INTERFACE_INCLUDE_DIRECTORIES)
+     target_include_directories(rocprofiler-sdk-elfio SYSTEM
+-- 
+2.52.0
+
-- 
2.43.0

